File: typescript-sdk/src/sdk/models/operations/postpublicauthorize.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PostPublicAuthorizePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=roomId" })
  roomId: string;
}


export class PostPublicAuthorizeRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PostPublicAuthorizePathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.PublicAuthorizeBodyRequest;
}


export class PostPublicAuthorizeResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  authorization?: shared.Authorization;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  error?: shared.ErrorT;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/postpublicauthorize.go
package operations

import (
	"openapi/pkg/models/shared"
)

type PostPublicAuthorizePathParams struct {
	RoomID string `pathParam:"style=simple,explode=false,name=roomId"`
}

type PostPublicAuthorizeRequest struct {
	PathParams PostPublicAuthorizePathParams
	Request    *shared.PublicAuthorizeBodyRequest `request:"mediaType=application/json"`
}

type PostPublicAuthorizeResponse struct {
	Authorization *shared.Authorization
	ContentType   string
	Error         *shared.Error
	StatusCode    int64
}
